fastlane_core/lib/fastlane_core/itunes_transporter.rb

```diff
@@ -180,9 +180,124 @@ def additional_upload_parameters
+      if Helper.test?
+        yield(nil) if block_given?
+        command
+      end
+
+      @errors = []
+      @warnings = []
+      @all_lines = []
+
+      if hide_output
+        # Show a one time message instead
+        UI.success("Waiting for App Store Connect transporter to be finished.")
+        UI.success("Application Loader progress... this might take a few minutes...")
+      end
+
+      begin
+        exit_status = FastlaneCore::FastlanePty.spawn(command) do |command_stdout, command_stdin, pid|
+          begin
+            command_stdout.each do |line|
+              @all_lines << line
+              parse_line(line, hide_output) # this is where the parsing happens
+            end
+          end
+        end
+      rescue => ex
+        # FastlanePty adds exit_status on to StandardError so every error will have a status code
+        exit_status = ex.exit_status
+        @errors << ex.to_s
+      end
+
+      @errors << "The call to the altool completed with a non-zero exit status: #{exit_status}. This indicates a failure." unless exit_status.zero?
+
+      UI.important(@warnings.join("\n")) unless @warnings.empty?
+
+      if @errors.count > 0 && @all_lines.count > 0
+        # Print out the last 18 lines, this is key for non-verbose mode
+        @all_lines.last(18).each do |line|
+          UI.important("[altool] #{line}")
+        end
+        UI.message("Application Loader output above ^")
+        UI.error(@errors.join("\n"))
+      end
+
+      yield(@all_lines) if block_given?
+      exit_status.zero?
+    end
+
+    def build_upload_command(username, password, source = "/tmp", provider_short_name = "", jwt = nil, platform = nil, api_key = nil)
+      use_api_key = !api_key.nil?
+      [
+        ("API_PRIVATE_KEYS_DIR=#{Dir.tmpdir}" if use_api_key),
+        "xcrun altool",
+        "--upload-app",
+        ("-u #{username.shellescape}" unless use_api_key),
+        ("-p #{password.shellescape}" unless use_api_key),
+        ("--apiKey #{api_key[:key_id]}" if use_api_key),
+        ("--apiIssuer #{api_key[:issuer_id]}" if use_api_key),
+        platform_option(platform),
+        file_upload_option(source),
+        additional_upload_parameters,
+        "-k 100000"
+      ].compact.join(' ')
+    end
+
+    def additional_upload_parameters
+      env_deliver_additional_params = ENV["DELIVER_ALTOOL_ADDITIONAL_UPLOAD_PARAMETERS"]
+      if env_deliver_additional_params.to_s.strip.empty?
+        return nil
+      end
+
+      return env_deliver_additional_params.to_s.strip
+    end
+
+    def handle_error(password)
+      UI.error("Could not download/upload from App Store Connect!")
+    end
+
+    def displayable_errors
+      @errors.map { |error| "[Application Loader Error Output]: #{error}" }.join("\n")
+    end
+
+    private
+
+    def file_upload_option(source)
+      "-f #{source.shellescape}"
+    end
+
+    def platform_option(platform)
+      "-t #{platform == 'osx' ? 'macox' : platform}"
```

Comment:
Typo?
```suggestion
      "-t #{platform == 'osx' ? 'macosx' : platform}"
```

Reply:
Btw, since that typo didn't make the test suite fail… maybe that's a sign that we need to add some test case (aka _spec_ example) to our test suite for it? 🤔

Reply:
also, this brings to point that are we really sure we need to make it `macosx` from `osx`? 

Apple's official document is saying `osx` should be enough which means we don't require this condition at all, right? maybe @freddi-kit knows? 👀 
https://help.apple.com/itc/apploader/#/apdATD1E53-D1E1A1303-D1E53A1126


<img width="1178" alt="Screenshot 2022-09-10 at 21 18 16" src="https://user-images.githubusercontent.com/5364500/189498752-a939ca47-6cdc-4188-b1f0-fbd9531abd5e.png">


> maybe that's a sign that we need to add some test case (aka spec example) to our test suite for it? 🤔

i agree!

Reply:
and when i m running `xcrun altool --help` in terminal then Apple is saying it should be `macos` 🙈 
so should we use `osx` or `macosx` or `macos`? 😅🤔 

To be honest, i have no mac-app to upload and verify about it...

<img width="1276" alt="Screenshot 2022-09-10 at 21 35 09" src="https://user-images.githubusercontent.com/5364500/189499227-00112356-35ac-4bdc-9eee-8d07340f2b97.png">

Reply:
This should be macos and it is my mistake 🙇 let me fix it

Reply:
> To be honest, i have no mac-app to upload and verify about it...

I just tested with new empty mac os app and it success to upload.

Reply:
https://github.com/fastlane/fastlane/pull/20631/commits/9988e33aecedff313164c4a441f80760a95d42f6 fixed